File: data/sample_evidence/VulnerableApp.java
// Deliberately vulnerable sample Java snippet — fixture data for
// core/owasp_security integration tests (pattern-based detection, since
// this project has no Java AST library — docs/adr/0021).
package com.example.demo;

import java.security.MessageDigest;
import java.util.Random;

public class VulnerableApp {
    public void runCommand(String userSuppliedCommand) throws Exception {
        Runtime.getRuntime().exec(userSuppliedCommand);
    }

    public String weakHash(String input) throws Exception {
        MessageDigest digest = MessageDigest.getInstance("MD5");
        return new String(digest.digest(input.getBytes()));
    }

    public int weakRandomToken() {
        Random random = new Random();
        return random.nextInt();
    }
}

File: data/sample_evidence/vulnerable_app.js
// Deliberately vulnerable sample Node.js snippet — fixture data for
// core/owasp_security integration tests (pattern-based detection, since
// this project has no JavaScript AST library — docs/adr/0021).
const child_process = require("child_process");

function runBackup(userSuppliedName) {
  child_process.exec("tar -cf backup.tar " + userSuppliedName);
}

function renderProfile(bio) {
  document.getElementById("bio").innerHTML = bio;
}

const apiKey = "sk-live-abcdef123456";

function generateToken() {
  return Math.random();
}

function unsafeEval(userInput) {
  eval(userInput);
}
